src/main.c

```diff
@@ -396,7 +398,7 @@ send_request_and_read_response(int sockfd,
         if(result == CRUSTLS_DEMO_AGAIN) {
           break;
         }
-        else if(result == CRUSTLS_DEMO_EOF) {
+        else if(result == CRUSTLS_DEMO_CLOSE_NOTIFY) {
```

Comment:
```suggestion
        else if(result == CRUSTLS_DEMO_CLOSE_NOTIFY || result == CRUSTLS_DEMO_EOF) {
```
`do_read` can still return `CRUSTLS_DEMO_EOF`, right?

Reply:
It can return EOF, but we actually want that to be an error. It falls through to the `else if(result != CRUSTLS_DEMO_OK)` on the next line.

It's an error because in theory we should always get `CRUSTLS_DEMO_CLOSE_NOTIFY` before we get an EOF.

Reply:
That makes sense. Then the remaining nit is that instead of exiting successfully on `CRUSTLS_DEMO_CLOSE_NOTIFY`, this program could instead enforce that it then gets `CRUSTLS_DEMO_EOF` in a timely manner.